plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal.hs

```diff
@@ -309,6 +317,50 @@ gotoDefinition ideState _ msgParam = do
       isSectionArgName name (Syntax.Section _ sectionArgName _) = name == CabalFields.onelineSectionArgs sectionArgName
       isSectionArgName _ _ = False
 
+-- | CodeActions for hover messages.
+--
+-- Provides a CodeAction for displaying message on hover.
+-- If found that the filtered hover message is a dependency,
+-- adds a Documentation link.
+hover :: PluginMethodHandler IdeState LSP.Method_TextDocumentHover
+hover ide _ msgParam = do
+      nfp <- getNormalizedFilePathE uri
+      (cabalFields, _) <- runActionE "cabal.cabal-hover" ide $ useWithStaleE ParseCabalFields nfp
+      let mCursorText = CabalFields.findTextWord cursor cabalFields
+      case mCursorText of
+        Nothing ->
+          pure $ InR Null
+        Just cursorText -> do
+          (gpd, _) <- runActionE "cabal.GPD" ide $ useWithStaleE ParseCabalFile nfp
+          let depsNames = map dependencyName $ allBuildDepends $ flattenPackageDescription gpd
+              mText = filterVersion cursorText
+          case mText of
+            Nothing -> pure $ foundHover (Nothing, [cursorText])
+            Just txt ->
+              if txt `elem` depsNames
+                then pure $ foundHover (Nothing, [txt <> "\n", documentationText txt])
+                else pure $ foundHover (Nothing, [txt])
+  where
+      cursor = Types.lspPositionToCabalPosition (msgParam ^. JL.position)
+      uri = msgParam ^. JL.textDocument . JL.uri
+
+      dependencyName :: Dependency -> T.Text
+      dependencyName dep = T.pack $ unPackageName $ depPkgName dep
+
+      filterVersion :: T.Text -> Maybe T.Text
+      filterVersion msg = getMatch (msg =~ regex)
+        where
+          regex :: T.Text
+          regex = "([a-zA-Z0-9-]*[a-zA-Z0-9]).*"
```

Comment:
```suggestion
          regex = "([a-zA-Z0-9-]+)"
```

I don't understand what the `.*` is for? You only care about the match group, right?

Reply:
> You only care about the match group, right?
Yes, I missed that, thank you.

Also can't replace `([a-zA-Z0-9-]*[a-zA-Z0-9]` with `[a-zA-Z0-9-]+`,
because the first one checks, it doesn't end with a `-`.

Reply:
Ah thanks! Perhaps add a comment then, because this is too easy to overlook!

Reply:
I'd suggest to make it `([-a-zA-Z0-9]*[a-zA-Z0-9]).*` then so that the difference stands out.

Reply:
@VenInf Would it be possible to apply this change?